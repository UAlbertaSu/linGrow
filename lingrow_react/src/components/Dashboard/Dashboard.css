
.form-control{
    margin: var(--margin-vsmall);
    width: fit-content;
}

.title_card{
    margin: var(--margin-vsmall);
    position: relative;
    left: auto;
    height: fit-content;
    width: fit-content;
    min-width: 250px;
    background: rgb(255, 255, 255);
    padding: 0px;
    margin-Top: 20px;
    margin-bottom: 10px;
}

.btn{
    margin: var(--margin-small);
    width: fit-content;
}
.error{
    margin: var(--margin-small);
    color: rgb(138, 0, 0);
    font-weight: bold;
}