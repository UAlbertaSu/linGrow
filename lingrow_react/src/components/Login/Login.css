
.form-control{
    margin: var(--margin-vsmall);
    width: fit-content;
}
.card {
    background-color: rgb(124, 194, 255);
    position: absolute;
    left:20%; 
    width: 60%; 
    min-width: none;
    display: flex;
    flex-direction: column;
    align-items: center;

}

.btn{
    margin: var(--margin-small);
    width: fit-content;
}
.error{
    margin: var(--margin-small);
    color: rgb(138, 0, 0);
    font-weight: bold;
}
@media only screen and (max-width: 800px) {
    [class*="card"]{
        width: 94%;
        left: 3%;
    }
}