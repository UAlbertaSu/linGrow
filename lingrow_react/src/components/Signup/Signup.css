:root {
    --margin-vsmall: 5px 5px 5px 5px;
    --margin-small: 10px 10px 10px 10px;
    --margin-medium: 20px 20px 20px 20px;
    --margin-large: 40px 40px 40px 40px;
}

h1 {
    font-size: 30px;
    font-weight: bold;
}

.label{
    display: block;
    font-size: larger;
    margin: var(--margin-vsmall);
    color: rgb(5, 5, 5);
}
.error{
    margin: var(--margin-small);
    color: rgb(204, 0, 0);
}
.success{
    margin: var(--margin-small);
    color: rgb(4, 202, 4);
}

.form-group{
    margin: var(--margin-small);
}

.card {
  position: absolute;
  width: fit-content;
  height: fit-content;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: aliceblue;
  padding: 2%;
  top: 40%;
  left: 50%;
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}