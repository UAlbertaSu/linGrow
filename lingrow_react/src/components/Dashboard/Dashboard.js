import React, { useEffect, useState, useCallback } from 'react';
import { useNavigate } from "react-router-dom";
import './Dashboard.css';
import { Card, Button} from 'react-bootstrap';
import {Helmet} from 'react-helmet';

import LanguageList from '../Translate/LanguageList';
import Translate from '../Translate/Translate';
import DashNav from '../DashNav/DashNav';
import logo from "../Img/blank_lingrow.png";



// A dashboard page that acts as the main navigation page for all users, 
// what is displayed is dependent on which user type is logged in.

export default function Dashboard({ userType }) {
    const nav = useNavigate();

    // State variables.
    const [tab_header, setTabHeader] = useState("LinGrow Home");
    const [chat, setChatMsg] = useState("Chat");
    const [manageSchools, setManageSchools] = useState("Manage Schools");
    const [manageUsers, setManageUsers] = useState("Manage Users");
    const [searchUsers, setSearchUsers] = useState("Search Users");
    const [group_manager, setGroupManagerMsg] = useState("Group Manager");
    const [activities, setLanguageLearningActivitiesMsg] = useState("Language Learning Activities");
    const [logout_msg, setLogoutMsg] = useState("Logout");

    // Navigate user to search user page.
   const searchUserHandler = async (event) => {
        event.preventDefault();
        nav('/searchuser');
   }

   const chatHandler = async (event) => {
        event.preventDefault();
        nav('/chat');
   }
   
    // Navigate user to language development activities page.
    const redirectToActivities = async (event) => {
        event.preventDefault();
        nav("/activities");
    }

    // Navigate user to login page if the token is not set or otherwise invalid.
    useEffect(() => {
        if (sessionStorage.getItem('token') === null || sessionStorage.getItem('token').includes("error")) {
            nav("/");
        }
    }, []);

    // Handle logout; clear session and navigate user to login page.
    const clearSession = async (event) => {
        sessionStorage.clear();
        nav("/");
    }

    // State variable for check if the page has been translated.
    const [translated, setTranslated] = useState(0);
    
    // Translate messages and set it to state variables.
    const translateMessage = useCallback((e) => {
        let lang = localStorage.getItem('lang');
        if (lang) {
            Translate('en', lang, "LinGrow Home").then((response) => setTabHeader(response));
            Translate('en', lang, "Chat").then(response => setChatMsg(response));
            Translate('en', lang, "Manage Schools").then(response => setManageSchools(response));
            Translate('en', lang, "Manage Users").then(response => setManageUsers(response));
            Translate('en', lang, "Search Users").then(response => setSearchUsers(response));
            Translate('en', lang, "Group Manager").then(response => setGroupManagerMsg(response));
            Translate('en', lang, "Language Learning Activities").then(response => setLanguageLearningActivitiesMsg(response));
            Translate('en', lang, "Logout").then(response => setLogoutMsg(response));
        }
    });

    // Run on the page load; translate page, and handle translate request when new language is set.
    useEffect(() => {
        if (!translated) {
            translateMessage();
            setTranslated(1);
        }
        
        window.addEventListener("New language set", translateMessage);
        return () => window.removeEventListener("New language set", translateMessage);
    }, []);


    // homepage with variables to display different content depending on user type.
    return (
        <div className="bg">
            <img src={logo}  class="center" alt="Lingrow Logo" style={{marginTop:"10px",marginBottom:"20px", maxHeight:"350px", maxWidth:"350px"}}/>
            <Card style={{paddingBottom:"10px", marginTop: "250px"}}>
                <Helmet>
                        <meta charSet="utf-8" />
                        <title>{tab_header}</title>
                </Helmet>
                <a href="https://bilingualacquisition.ca/"></a>
                <LanguageList />
                <DashNav/>
                <Card className='bg-light' style={{position:"relative", left:"0%", marginBottom:"15px", width:"94%", padding:"25px"}}>
                    <Button href="chat" variant="primary" type="submit" id="chat" style={{minWidth:"150px"}}>{chat}</Button>  
                    <div>{userType === 4 ? <Button href="/schoolmanager" variant="primary" type="submit" id="manageSchools" style={{minWidth:"150px"}}>{manageSchools}</Button> : null}</div>
                    <div>{userType > 1 ? <Button variant="primary" type="submit" id="searchUsers" onClick = {searchUserHandler}style={{minWidth:"150px"}}>{searchUsers}</Button> : null}</div>
                    <div>{userType === 4 ? <Button href="/usermanager" variant="primary" type="submit" id="manageUsers" style={{minWidth:"150px"}}>{manageUsers}</Button> : null}</div>
                    <div>{userType !== 1 ? <Button href="/groupmanager" variant="primary" type="submit" id="groups" style={{minWidth:"150px"}}>{group_manager}</Button> : null}</div>  
                    <Button variant="secondary" type="submit" id="activities" onClick={redirectToActivities} style={{minWidth:"150px"}}>{activities}</Button>
                    <Button variant="danger" type="submit" id="logout" onClick={clearSession} style={{minWidth:"150px"}}>{logout_msg}</Button>
                </Card>
            </Card>
        </div>
    );
}