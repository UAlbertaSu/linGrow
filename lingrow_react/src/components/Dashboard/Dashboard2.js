import React, { useEffect } from 'react';
import { useNavigate } from "react-router-dom";
import './Dashboard.css';
import { Card } from 'react-bootstrap';

export default function Dashboard() {
    const nav = useNavigate();

    useEffect(() => {
        if (sessionStorage.getItem('token') === null || sessionStorage.getItem('token').includes("error")) {
            nav("/");
        }
    }, []);

    const clearSession = async (event) => {
        sessionStorage.clear();
        sessionStorage.setItem('redirect', "success");
        nav("/");
    }

    const teacher_dashboard = () => {
        // ...
    }

    const parent_dashboard = () => {
        // ...
    }

    return (
        <div className="dashboard-wrapper">
            <Card>
                <h2>Welcome to LinGrow dashboard</h2>
                <div>
                    {teacher_dashboard()}
                    {parent_dashboard()}
                </div>
                <button onClick={clearSession} data-testid='logout'>Logout</button>
                <a href="https://drive.google.com/drive/folders/1h4pmfp66la3ZBpEIwcfHb7TEY5QbUgOj">Kitchen Activities</a>
                <a href="https://drive.google.com/drive/folders/1Pbaax2cLWvOSO8sY2Lm8by0lE0G8njRJ">Bath Time!</a>
            </Card>
        </div>
    );
}