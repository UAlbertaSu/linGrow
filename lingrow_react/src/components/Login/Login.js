import React, { useState } from 'react';
import { useNavigate } from "react-router-dom";
import { Form, Button, Card } from 'react-bootstrap';

import logo from "../Img/lingrow.png";
import './Login.css';

async function loginUser(credentials) {
    console.log(JSON.stringify(credentials));
    return fetch('http://127.0.0.1:8000/api/user/login/', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        body: JSON.stringify(credentials)
    }).then(data => data.json())
}

export default function Login() {
    const nav = useNavigate();
    const[email, setEmail] = useState();
    const[password, setPassWord] = useState();
    const[error, setError] = useState(false);


    const handleSubmit = async (event) =>{
        event.preventDefault();
        const token = await loginUser({
            "email": email,
            "password": password
        });
        sessionStorage.setItem('token', JSON.stringify(token));

        if (sessionStorage.getItem('token').includes("error")) {
            setError(true);
            console.log("Validation failed");
        }
        else {
            setError(false);
            console.log("Valid credentials");
            nav("/dashboard");
        }
    }

    const redirectToSignup = async (event) => {
        event.preventDefault();
        nav("/signup");
    }

    const errorMessage = () => {
        return (
            <div className="error" style={{
            display: error ? '' : 'none',
            }}>
                Invalid email or password
            </div>
        );
    };

    return (
        <Card>
            <img src={logo} className="logo" alt="logo" />
            <h1>LinGrow Login</h1>
            <label className="label">Email address</label>
            <input type="text" class="form-control" id="name" placeholder="Enter Email Address" onChange={e => setEmail(e.target.value)}/>
            <label className="label">Password</label>
            <input type="password" class="form-control" id="password" placeholder="Enter Password" onChange={e => setPassWord(e.target.value)}/>
            <div className="message">
                {errorMessage()}
            </div>
            <Button variant="primary" type="submit" onClick={handleSubmit}>Login</Button>
            <Button variant="primary" type="submit" onClick={redirectToSignup}>Sign Up</Button>
            <a href="https://drive.google.com/drive/folders/1h4pmfp66la3ZBpEIwcfHb7TEY5QbUgOj">Kitchen Activities</a>
            <a href="https://drive.google.com/drive/folders/1Pbaax2cLWvOSO8sY2Lm8by0lE0G8njRJ">Bath Time!</a>
        </Card>
    )
}