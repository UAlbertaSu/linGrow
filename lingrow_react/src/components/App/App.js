import { BrowserRouter, Routes, Route, Navigate } from 'react-router-dom';

import './App.css';
import Login from '../Login/Login';
import DashboardWrapper from '../Dashboard/DashboardWrapper';
import Signup from '../Signup/Signup';
import Welcome from '../Welcome/Welcome';
import Activities from '../Activities/Activities';
import UserInfoWrapper from '../UserInfo/UserInfoWrapper';
import GroupManagerWrapper from '../GroupManager/GroupManagerWrapper';
import GroupCreatorWrapper from '../GroupManager/GroupCreatorWrapper';
import GroupDetail from '../GroupManager/GroupDetail';
import UserSearch from '../UserSearch/UserSearch';
import AdminAddUser from '../Signup/AdminAddUser';
import UserManagerWrapper from '../UserManager/UserManagerWrapper';
import SchoolManager from '../SchoolManager/SchoolManager';
import SchoolCreator from '../SchoolManager/SchoolCreator';
import ClassroomManager from '../ClassroomManager/ClassroomManager';
import ClassroomDetail from '../ClassroomManager/ClassroomDetail';
import ClassroomCreator from '../ClassroomManager/ClassroomCreator';

function App() {
  if (localStorage.getItem('lang') === null) {
    return <Welcome />
  }

  return (
    <div className="wrapper">
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Navigate to="/login" />}></Route>
          <Route path="/dashboard" element={<DashboardWrapper/ >}></Route>
          <Route path="/welcome" element={<Welcome/ >}></Route>
          <Route path="/userinfo" element={<UserInfoWrapper/ >}></Route>
          <Route path="/groupmanager" element={<GroupManagerWrapper/ >}></Route>
          <Route path="/usermanager" element={<UserManagerWrapper/ >}></Route>
          <Route path="/groupcreator" element={<GroupCreatorWrapper/ >}></Route>
          <Route path="/adminadduser" element={<AdminAddUser/ >}></Route>
          <Route path="/groupdetail" element={<GroupDetail/ >}></Route>
          <Route path="/login" element={<Login/ >}></Route>
          <Route path="/signup" element={<Signup/ >}></Route>
          <Route path="/activities" element={<Activities/ >}></Route>
          <Route path = "/searchuser" element= {<UserSearch />}> </Route>
          <Route path="/chat"></Route>
          <Route path="/schoolmanager" element={<SchoolManager/ >}></Route>
          <Route path="/schoolcreator" element={<SchoolCreator/ >}></Route>
          <Route path="/classroommanager" element={<ClassroomManager/ >}></Route>
          <Route path="/classroomdetail" element={<ClassroomDetail/ >}></Route>
          <Route path="/classroomcreator" element={<ClassroomCreator/ >}></Route>

        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
