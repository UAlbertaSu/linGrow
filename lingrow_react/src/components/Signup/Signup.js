import React, { useState, useCallback, useEffect} from 'react';
import { useNavigate } from "react-router-dom";

import LanguageList from '../Translate/LanguageList';
import Translate from '../Translate/Translate';

import './Signup.css';
import 'react-bootstrap/Form'
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button'

export default function Form() {

    const nav = useNavigate();

    //translation states
    const [header, setHeader] = useState("User Registration");
    const [email_msg, setEmailMsg] = useState("Email address");
    const [enter_email_msg, setEnterEmailMsg] = useState("Enter Email Address");
    const [first_msg, setFirstMsg] = useState("First name");
    const [enter_first_msg, setEnterFirstMsg] = useState("Enter First Name");
    const [middle_msg, setMiddleMsg] = useState("Middle name");
    const [enter_middle_msg, setEnterMiddleMsg] = useState("Enter Middle Name");
    const [last_msg, setLastMsg] = useState("Last name");
    const [enter_last_msg, setEnterLastMsg] = useState("Enter Last Name");
    const [type_msg, setTypeMsg] = useState("User type");
    const [enter_type_msg, setEnterTypeMsg] = useState("Please select User Type");
    const [parent_msg, setParentMsg] = useState("Parent");
    const [teacher_msg, setTeacherMsg] = useState("Teacher");
    const [researcher_msg, setResearcherMsg] = useState("Researcher");
    const [admin_msg, setAdminMsg] = useState("Admin");
    const [pass_msg, setPasswordMsg] = useState("Password");
    const [enter_pass_msg, setEnterPasswordMsg] = useState("Enter password");
    const [confirm_msg, setConfirmPasswordMsg] = useState("Confirm password");
    const [enter_confirm_msg, setEnterConfirmPasswordMsg] = useState("Enter password again");
    const [child_msg, setChildMsg] = useState("Child's name(* if parent)");
    const [enter_child_msg, setEnterChildMsg] = useState("Enter Child's Name");
    const [error_msg, setErrorMsg] = useState("Registration failed");
    const [require_msg, setRequiredMsg] = useState("Field with * must be filled out");
    const [register_btn, setRegisterBtn] = useState("Register");


    async function signupUser(credentials) {
        console.log(credentials);
        // sessionStorage.clear();
        // sessionStorage.setItem('registration', "success");
        return fetch('http://127.0.0.1:8000/api/user/register/', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(credentials)
        }).then(data => {
            console.log(data);
            if (data['status'] === 201) {
                // sessionStorage.clear();
                // sessionStorage.setItem('registration', "success");
                setSubmitted(true);
                setError(false);
                nav("/login");
            }
            else {
                // sessionStorage.clear();
                // sessionStorage.setItem('registration', "failed");
                setSubmitted(false);
                setError(true);
            }
        })
    }

    // States for registration
    const [email, setEmail] = useState('');
    const [first_name, setFirstName] = useState('');
    const [last_name, setLastName] = useState('');
    const [user_type, setUserType] = useState('');
    const [password, setPassword] = useState('');
    const [password2, setPassword2] = useState('');
    const [child_name, setChildName] = useState('');
    const [middle_name, setMiddleName] = useState('');

    // States for checking the errors
    const [submitted, setSubmitted] = useState(false);
    const [error, setError] = useState(false);

    // Handling the parameter changes
    const handleEmail = (e) => {
        setEmail(e.target.value);
        setSubmitted(false);
    };
    
    const handleFirstName = (e) => {
        setFirstName(e.target.value);
        setSubmitted(false);
    };
    
    const handleLastName = (e) => {
        setLastName(e.target.value);
        setSubmitted(false);
    };
    
    const handleUserType = (e) => {
        setUserType(e.target.value);
        setSubmitted(false);
    };
    
    const handlePassword = (e) => {
        setPassword(e.target.value);
        setSubmitted(false);
    };
    
    const handleConfirmPassword = (e) => {
        setPassword2(e.target.value);
        setSubmitted(false);
    };

    const handleChildName = (e) => {
        setChildName(e.target.value);
        setSubmitted(false);
    };

    const handleMiddleName = (e) => {
        setMiddleName(e.target.value);
        setSubmitted(false);
    };

    // Handling the form submission
    const handleSubmit = (e) => {

        e.preventDefault();
        if (email === '' || first_name === '' || last_name === '' || user_type === '' || password === '' || password2 === '') {
            setError(true);
            // sessionStorage.clear();
            // sessionStorage.setItem('registration', "failed");
        }
        else if (password !== password2) {
            setError(true);
            // sessionStorage.clear();
            // sessionStorage.setItem('registration', "failed");
        }
        else if (user_type === '1' && child_name === '') {
            setError(true);
            // sessionStorage.clear();
            // sessionStorage.setItem('registration', "failed");
        }
        else {
            e.preventDefault();
            signupUser({
                "email": email,
                "first_name": first_name,
                "last_name": last_name,
                "user_type": user_type,
                "password": password,
                "password2": password2,
                "child_name": child_name,
                "middle_name": middle_name
            });

            // // TODO: Would be nice to have a timed redirect to dashboard, rather than immediate redirect.
            // nav("/dashboard");
        }
    };

    const [translated, setTranslated] = useState(0);

    // Showing success message
    const successMessage = () => {
        return (
            <div className="success" style={{
                display: submitted ? '' : 'none',
            }}>
                Welcome to linGrow {first_name} {last_name}!
            </div>
        );
    };

    const translateMessage = useCallback((e) => {
        let lang = localStorage.getItem('lang');
        if (lang) {
            Translate('en', lang, "User Registration").then(response => setHeader(response));
            Translate('en', lang, "Email address*").then(response => setEmailMsg(response));
            Translate('en', lang, "Enter Email Address").then(response => setEnterEmailMsg(response));
            Translate('en', lang, "First name*").then(response => setFirstMsg(response));
            Translate('en', lang, "Enter First Name").then(response => setEnterFirstMsg(response));
            Translate('en', lang, "Middle name").then(response => setMiddleMsg(response));
            Translate('en', lang, "Enter Middle Name").then(response => setEnterMiddleMsg(response));
            Translate('en', lang, "Last name*").then(response => setLastMsg(response));
            Translate('en', lang, "Enter Last Name").then(response => setEnterLastMsg(response));
            Translate('en', lang, "User type*").then(response => setTypeMsg(response));
            Translate('en', lang, "Please select User Type").then(response => setEnterTypeMsg(response));
            Translate('en', lang, "Parent").then(response => setParentMsg(response));
            Translate('en', lang, "Teacher").then(response => setTeacherMsg(response));
            Translate('en', lang, "Researcher").then(response => setResearcherMsg(response));
            Translate('en', lang, "Admin").then(response => setAdminMsg(response));
            Translate('en', lang, "Please Select User Type").then(response => setEnterTypeMsg(response));
            Translate('en', lang, "Password*").then(response => setPasswordMsg(response));
            Translate('en', lang, "Enter Password").then(response => setEnterPasswordMsg(response));
            Translate('en', lang, "Confirm password*").then(response => setConfirmPasswordMsg(response));
            Translate('en', lang, "Enter password again").then(response => setEnterConfirmPasswordMsg(response));
            Translate('en', lang, "Child's Name(* if parent)").then(response => setChildMsg(response));
            Translate('en', lang, "Enter Child's Name").then(response => setEnterChildMsg(response));
            Translate('en', lang, "Error").then(response => setErrorMsg(response));
            Translate('en', lang, "Field with * is required").then(response => setRequiredMsg(response));
            Translate('en', lang, "Register").then(response => setRegisterBtn(response));
        }
    });

    useEffect(() => {
        // Prevents page from being constantly translated.
        if (!translated) {
            translateMessage();
            setTranslated(1);
        }

        window.addEventListener("New language set", translateMessage);
        return () => window.removeEventListener("New language set", translateMessage);
    });

    // Showing error message if error is true
    const errorMessage = () => {
        return (
            <div className="error" id="errormessage" style={{
            display: error ? '' : 'none',
            }}>
                {error_msg}
            </div>
        );
    };

    return (
        <Card style={{position:"relative",height:"fit-content", maxWidth:"fit-content", left:"40%"}}>
                    <LanguageList />
                    <h1>{header}</h1>
                        {/* Labels and inputs for form data */}
                    <label className="label">{email_msg}</label>
                    <input className="form-control" type="text" id="email" placeholder={enter_email_msg} value={email} onChange={handleEmail} />
                    <label className="label">{first_msg}</label>
                    <input className="form-control" type="text" id="first_name" placeholder={enter_first_msg} value={first_name} onChange={handleFirstName} />
                    <label className="label">{middle_msg}</label>
                    <input className="form-control" type="text" id="middle_name" placeholder={enter_middle_msg} value={middle_name} onChange={handleMiddleName} />
                    <label className="label">{last_msg}</label>
                    <input className="form-control" type="text" id="last_name" placeholder={enter_last_msg} value={last_name} onChange={handleLastName} />
                    <label className="label">{type_msg}</label>
                    <select className="form-select" value={user_type} id="user_type" onChange={handleUserType}>
                        <option value="" disabled selected>{enter_type_msg}</option>
                        <option value="1">{parent_msg}</option>
                        <option value="2">{teacher_msg}</option>
                        <option value="3">{researcher_msg}</option>
                        <option value="4">{admin_msg}</option>
                    </select>
                    <label className="label">{pass_msg}</label>
                    <input className="form-control" type="password" id="password1" placeholder={enter_pass_msg} value={password} onChange={handlePassword} />
                    <label className="label">{confirm_msg}</label>
                    <input className="form-control" type="password" id="password2" placeholder={enter_confirm_msg} value={password2} onChange={handleConfirmPassword} />
                    <label className="label">{child_msg}</label>
                    <input className="form-control" type="text" id="child_name" placeholder={enter_child_msg} value={child_name} onChange={handleChildName} />
                    <label classname="label" >{require_msg}</label>
                    <div className="message">
                        {errorMessage()}
                        {successMessage()}
                    </div>
                    <Button variant="primary" type="submit" id="submit_button" onClick={handleSubmit}>{register_btn}</Button>{''}
        </Card>
    );
}