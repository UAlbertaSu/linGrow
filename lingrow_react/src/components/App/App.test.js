import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import App from './App';
import Login from '../Login/Login';
import Dashboard from '../Dashboard/Dashboard';
import Signup from '../Signup/Signup';

// Test each page renders correctly.
describe("Snapshot testing of frontend", () => {
  test('Render login page', () => {
    const linkElement = renderer.create(
      <BrowserRouter>
        <Login />
      </BrowserRouter>).toJSON();
    expect(linkElement).toMatchSnapshot();
  });

  test('Render signup page', () => {
    const linkElement = renderer.create(
      <BrowserRouter>
        <Signup />
      </BrowserRouter>).toJSON();
    expect(linkElement).toMatchSnapshot();
  });

  test('Render dashboard', () => {
    const linkElement = renderer.create(
      <BrowserRouter>
        <Dashboard />
      </BrowserRouter>).toJSON();
    expect(linkElement).toMatchSnapshot();
  });
});

describe("Link functionality tests", () => {
  test('Test that default page / redirects to the Login', () => {
    render(<App />);
    expect(global.window.location.pathname).toContain('/login');
  });

  test('Test activity links from login page', () => {
    render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );
    const linkEl = screen.getByText('Kitchen Activities');
    expect(linkEl).toHaveAttribute("href", "https://drive.google.com/drive/folders/1h4pmfp66la3ZBpEIwcfHb7TEY5QbUgOj");
  });

  test('Test activity links from dashboard', () => {
    render(
      <BrowserRouter>
        <Dashboard />
      </BrowserRouter>
    );
    const linkEl = screen.getByText('Kitchen Activities');
    expect(linkEl).toHaveAttribute("href", "https://drive.google.com/drive/folders/1h4pmfp66la3ZBpEIwcfHb7TEY5QbUgOj");
  });

  test('Test logout button from dashboard navigates to the logout page', () => {
    // ???
  });

  test('Test signup button in login page navigates to the signup page', () => {
    // ???
  });
});

describe("Check login functionality", () => {
  test('Test login fails without any credentials', () => {
    render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );
    userEvent.click(screen.getByText("Login"));
    expect(screen.getByText("Invalid email or password")).toBeInTheDocument();
  });

  test('Test login with valid credential', () => {
    // ???
  });

  test('Login with valid credential, logout, and immediately attempt login without valid credential', () => {
    // ???
  });
});

describe("Check signup functionality", () => {

})