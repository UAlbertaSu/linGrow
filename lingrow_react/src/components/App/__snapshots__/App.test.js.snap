// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing of frontend Render dashboard 1`] = `
<div
  className="dashboard-wrapper"
>
  <div
    className="card"
  >
    <h2>
      Welcome to LinGrow dashboard
    </h2>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
    <a
      href="https://drive.google.com/drive/folders/1h4pmfp66la3ZBpEIwcfHb7TEY5QbUgOj"
    >
      Kitchen Activities
    </a>
    <a
      href="https://drive.google.com/drive/folders/1Pbaax2cLWvOSO8sY2Lm8by0lE0G8njRJ"
    >
      Bath Time!
    </a>
  </div>
</div>
`;

exports[`Snapshot testing of frontend Render login page 1`] = `
<div
  className="card"
>
  <img
    alt="logo"
    className="logo"
    src="lingrow.png"
  />
  <h1>
    LinGrow Login
  </h1>
  <label
    className="label"
  >
    Email address
  </label>
  <input
    class="form-control"
    id="name"
    onChange={[Function]}
    placeholder="Enter Email Address"
    type="text"
  />
  <label
    className="label"
  >
    Password
  </label>
  <input
    class="form-control"
    id="password"
    onChange={[Function]}
    placeholder="Enter Password"
    type="password"
  />
  <div
    className="message"
  >
    <div
      className="error"
      style={
        Object {
          "display": "none",
        }
      }
    >
      Invalid email or password
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    Login
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    Sign Up
  </button>
  <a
    href="https://drive.google.com/drive/folders/1h4pmfp66la3ZBpEIwcfHb7TEY5QbUgOj"
  >
    Kitchen Activities
  </a>
  <a
    href="https://drive.google.com/drive/folders/1Pbaax2cLWvOSO8sY2Lm8by0lE0G8njRJ"
  >
    Bath Time!
  </a>
</div>
`;

exports[`Snapshot testing of frontend Render signup page 1`] = `
<div
  className="card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <div
    className="form"
  >
    <form
      className="input"
    >
      <div>
        <h1>
          User Registration
        </h1>
        <label
          className="label"
        >
          Email address
        </label>
        <input
          id="email"
          onChange={[Function]}
          placeholder="Enter Email Address"
          type="text"
          value=""
        />
        <label
          className="label"
        >
          First name
        </label>
        <input
          id="first_name"
          onChange={[Function]}
          placeholder="Enter First Name"
          type="text"
          value=""
        />
        <label
          className="label"
        >
          Middle name
        </label>
        <input
          id="middle_name"
          onChange={[Function]}
          placeholder="Enter Middle Name"
          type="text"
          value=""
        />
        <label
          className="label"
        >
          Last name
        </label>
        <input
          id="last_name"
          onChange={[Function]}
          placeholder="Enter Last Name"
          type="text"
          value=""
        />
        <label
          className="label"
        >
          User type
        </label>
        <select
          onChange={[Function]}
          value=""
        >
          <option
            value="0"
          >
            Select User Type
          </option>
          <option
            value="1"
          >
            Parent
          </option>
          <option
            value="2"
          >
            Teacher
          </option>
          <option
            value="3"
          >
            Researcher
          </option>
          <option
            value="4"
          >
            Admin
          </option>
        </select>
        <label
          className="label"
        >
          Password
        </label>
        <input
          id="password"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
          value=""
        />
        <label
          className="label"
        >
          Confirm password
        </label>
        <input
          id="password2"
          onChange={[Function]}
          placeholder="Confirm Password"
          type="password"
          value=""
        />
        <label
          className="label"
        >
          Child name
        </label>
        <input
          id="name"
          onChange={[Function]}
          placeholder="Enter Child Name"
          type="text"
          value=""
        />
      </div>
      <div
        className="message"
      >
        <div
          className="error"
          style={
            Object {
              "display": "none",
            }
          }
        >
          Registration failed
        </div>
        <div
          className="success"
          style={
            Object {
              "display": "none",
            }
          }
        >
          Welcome to linGrow 
           
          !
        </div>
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
